### PR TITLE
drivers: migrate xtimer64 and xtimer/ticks users to ztimer

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR takes care of the user's of xtimer which can't be directly changed to ztimer with the coccinelle script.

### Testing procedure

I don't have any of the hardware sadly... but I tagged people that are susceptible to having the hardware, it will at least document what was and not tested.

- [ ]  `tests/driver_pir` @maribu ?
- [ ]  `tests/driver_si1133` @iosabi  ?
- [ ] `tests/driver_nrf24l01p` @PeterKietzmann  ?
- [ ] `tests/driver_mtd_nvram_spi` AFAIK there is one `mulle` in FU so maybe @miri64 ? or @jnohlgard?
- [ ] `tests/driver_ltc4150` @maribu 

### Issues/PRs references

Part of #13667 and #17111

### CODE REVIEW

**Comment by maribu** [comment]
Regarding the ltc coulomb counter: I think it makes sense to use `ztimer` (not 64) with milliseconds precision instead. Chances are, that the coulomb counter is used to monitor the power consumption of the board attached to it, so one might be interested in using low power. Also, microsecond resolution is likely not needed, as with typical current consumption there is at least roughly a second between to pulses.

I think the ltc driver is one of my first drivers, so likely it is shitty noob code. Maybe I should take a look at it again, clean it up, and port it to ztimer?

**Comment by fjmolinas** [comment]
> I think the ltc driver is one of my first drivers, so likely it is shitty noob code. Maybe I should take a look at it again, clean it up, and port it to ztimer?

As you prefer!

**Comment by fjmolinas** [comment]
> > I think the ltc driver is one of my first drivers, so likely it is shitty noob code. Maybe I should take a look at it again, clean it up, and port it to ztimer?
> 
> As you prefer!

@maribu should I split that one out and you open a separate PR for it?

**Comment by fjmolinas** [comment]
> > > I think the ltc driver is one of my first drivers, so likely it is shitty noob code. Maybe I should take a look at it again, clean it up, and port it to ztimer?
> > 
> > 
> > As you prefer!
> 
> @maribu should I split that one out and you open a separate PR for it?

Otherwise we get this one in (quickly get rid of xtimer) and then you can do the re-work?

**Comment by fjmolinas** [comment]
Rebased to fix conflict

**Comment by fjmolinas** [comment]
All green @maribu if you would rather still split out the ltc, please feel free to force push to the branch, I'll be AFK ideally for the last month.

File: tests/driver_ltc4150/main.c [hunk 1]
```diff
@@ -96,8 +97,8 @@ static void pulse_cb(ltc4150_dev_t *dev, ltc4150_dir_t dir, uint64_t now_usec,
  */
 static void spin(uint32_t seconds)
 {
-    uint32_t till = xtimer_now_usec() + US_PER_SEC * seconds;
-    while (xtimer_now_usec() < till) { }
+    uint32_t till = ztimer_now(ZTIMER_USEC) + US_PER_SEC * seconds;
+    while (ztimer_now(ZTIMER_USEC) < till) { }
```
Comment (kfessel) [inline tests/driver_ltc4150/main.c:101]: overflow might occur -the old code has the same problem-; works due to short test run time < 1h:11min 

may use ztimer_spin  instead

File: tests/driver_pir/main.c [hunk 2]
```diff
@@ -76,7 +77,7 @@ int main(void)
     while (1) {
         printf("Status: %s\n", pir_get_status(&dev) == PIR_STATUS_INACTIVE ?
                "inactive" : "active");
-        xtimer_usleep(1000 * 1000);
+        ztimer_sleep(ZTIMER_USEC, US_PER_SEC);
```
Comment (kfessel) [inline tests/driver_pir/main.c:80]:
```suggestion
        ztimer_sleep(ZTIMER_USEC, 1 * US_PER_SEC);
```

Comment (kfessel) [reply tests/driver_pir/main.c:80]: may as well be moved to a follow up

Comment (fjmolinas) [reply tests/driver_pir/main.c:80]: Is this worth running the ci again?

File: tests/driver_ltc4150/main.c (hunk 1 above)
Comment (fjmolinas) [reply tests/driver_ltc4150/main.c:101]: This makes sense to include

File: tests/driver_pir/main.c (hunk 2 above)
Comment (fjmolinas) [reply tests/driver_pir/main.c]: Addressed

File: tests/driver_ltc4150/main.c (hunk 1 above)
Comment (fjmolinas) [reply tests/driver_ltc4150/main.c]: done

**Comment by fjmolinas** [comment]
@maribu can we get this in?

File: tests/events/modules.kconfig.txt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+periph_pm
+periph_timer
+periph_uart
+stdin
+stdio_uart
+stdio_uart_rx
+sys
+test_utils_interactive_sync
+tsrb
+xtimer
```
Comment (maribu) [inline tests/events/modules.kconfig.txt:41]: This looks like a debugging artifact to me

File: tests/events/modules.make.txt [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+xtimer_on_ztimer
+ztimer
+ztimer_convert
+ztimer_convert_frac
+ztimer_convert_shift
+ztimer_core
+ztimer_extend
+ztimer_init
+ztimer_periph_timer
+ztimer_usec
```
Comment (maribu) [inline tests/events/modules.make.txt:66]: Same

**Review by maribu — COMMENTED** [review]
Code looks good to me. I don't have time for testing, but I also don't see anything that would break due to the switch, unless this would trigger an unknown bug in ztimer.

File: tests/events/modules.make.txt (hunk 4 above)
Comment (fjmolinas) [reply tests/events/modules.make.txt]: Thanks!

File: tests/events/modules.kconfig.txt (hunk 3 above)
Comment (fjmolinas) [reply tests/events/modules.kconfig.txt]: Thanks!

**Comment by fjmolinas** [comment]
> Code looks good to me. I don't have time for testing, but I also don't see anything that would break due to the switch, unless this would trigger an unknown bug in ztimer.

Yes, at least its explicit that not all was tested, and its pretty isolated code. Only isolated driver code.

**Review by maribu — APPROVED** [review]
The code looks good an unscary. If anyone insist on testing every driver, speak up now ;)

**Comment by fjmolinas** [comment]
> The code looks good an unscary. If anyone insist on testing every driver, speak up now ;)

Lets go then!

**Comment by fjmolinas** [comment]
Thanks for the review!

**Pull request merged**